packages/flutter_tools/lib/src/base/clock.dart

```diff
@@ -0,0 +1,16 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:quiver/time.dart';
+
+import 'context.dart';
```

Comment:
Done. Removed `Clock` from `AppContext`. Personally I have not experienced issues with the `AppContext` approach. Like with many dependency injection frameworks it has the disadvantage of making dependencies implicit. But in return it reduces some boilerplate. It's a trade-off I'm used to.